### PR TITLE
Upload photos

### PR DESCRIPTION
**Pull request by Crash1212**

### CODE REVIEW

File: modules/upload/upload.service.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    process.env.AZURE_HOST
+);
+const containerName = 'images';
+const getStream = require('into-stream');
+const Jimp = require("jimp");
+const uniqid = require('uniqid');
+
+class UploadService {
+    async uploadFiles(files) {
+        return files.map(async file => {
```
Comment (Alan18081) [inline modules/upload/upload.service.js:15]: Please, use here `Promise.all` to return response only when all files are uploaded

File: modules/upload/upload.service.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            const { createReadStream, filename } = await file;
```
Comment (Alan18081) [inline modules/upload/upload.service.js:16]: Do we need an `await` here? Is `file` a promise?

File: modules/upload/upload.service.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+                    fileBuffer = Buffer.concat(chunks);
+                    return resolve(fileBuffer);
+                });
+
+                inputStream.on('data', (chunk) => {
+                    chunks.push(chunk);
+                });
+            })
+
+            Jimp.read(inputBuffer).then(image => {
```
Comment (Alan18081) [inline modules/upload/upload.service.js:38]: Use `async await` instead of `then`

File: modules/upload/upload.service.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+                        streamLength = buffer.length
+
+
+                    blobService.createBlockBlobFromStream(containerName, blobName, stream, streamLength, err => {
+                        if (err) {
+                            throw new Error(err)
+                        }
+                    })
+                })
+                image.resize(1080, Jimp.AUTO).getBuffer(image.getMIME(), (err, buffer) => {
```
Comment (Alan18081) [inline modules/upload/upload.service.js:52]: Please, convert `resize` to `Promise` using `promisify` from the default node package `util`

File: modules/upload/upload.service.js [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+                    return resolve(fileBuffer);
+                });
+
+                inputStream.on('data', (chunk) => {
+                    chunks.push(chunk);
+                });
+            })
+
+            Jimp.read(inputBuffer).then(image => {
+                image.resize(1920, Jimp.AUTO).getBuffer(image.getMIME(), (err, buffer) => {
```
Comment (Alan18081) [inline modules/upload/upload.service.js:39]: Please, convert `resize` to `Promise` using `promisify` from the default node package `util`

File: modules/upload/upload.service.js [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+                    const
+                        blobName = `large_${id}_${filename}`,
+                        stream = getStream(buffer),
+                        streamLength = buffer.length
+
+
+                    blobService.createBlockBlobFromStream(containerName, blobName, stream, streamLength, err => {
```
Comment (Alan18081) [inline modules/upload/upload.service.js:46]: Please, convert `createBlockBlobFromStream` to `Promise` using `promisify` from the default node package `util`

File: modules/upload/upload.service.js [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+                        }
+                    })
+                })
+                image.resize(1080, Jimp.AUTO).getBuffer(image.getMIME(), (err, buffer) => {
+                    const
+                        blobName = `medium_${id}_${filename}`,
+                        stream = getStream(buffer),
+                        streamLength = buffer.length
+
+                    blobService.createBlockBlobFromStream(containerName, blobName, stream, streamLength, err => {
```
Comment (Alan18081) [inline modules/upload/upload.service.js:58]: Please, convert `createBlockBlobFromStream` to `Promise` using `promisify` from the default node package `util`

File: modules/upload/upload.service.js [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+                        if (err) {
+                            throw new Error(err)
+                        }
+                    })
+                })
+
+                image.resize(768, Jimp.AUTO).getBuffer(image.getMIME(), (err, buffer) => {
```
Comment (Alan18081) [inline modules/upload/upload.service.js:65]: Please, convert `resize` to `Promise` using `promisify` from the default node package `util`

File: modules/upload/upload.service.js [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+                    const
+                        blobName = `small_${id}_${filename}`,
+                        stream = getStream(buffer),
+                        streamLength = buffer.length
+
+                    blobService.createBlockBlobFromStream(containerName, blobName, stream, streamLength, err => {
```
Comment (Alan18081) [inline modules/upload/upload.service.js:71]: Please, convert `createBlockBlobFromStream` to `Promise` using `promisify` from the default node package `util`

File: modules/upload/upload.service.js [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+                        if (err) {
+                            throw new Error(err)
+                        }
+                    })
+                })
+
+                image.resize(128, Jimp.AUTO).getBuffer(image.getMIME(), (err, buffer) => {
```
Comment (Alan18081) [inline modules/upload/upload.service.js:78]: Please, convert `resize` to `Promise` using `promisify` from the default node package `util`

File: modules/upload/upload.service.js [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+                    const
+                        blobName = `thumbnail_${id}_${filename}`,
+                        stream = getStream(buffer),
+                        streamLength = buffer.length
+
+                    blobService.createBlockBlobFromStream(containerName, blobName, stream, streamLength, err => {
```
Comment (Alan18081) [inline modules/upload/upload.service.js:84]: Please, convert `createBlockBlobFromStream` to `Promise` using `promisify` from the default node package `util`

File: modules/upload/upload.service.js [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+                    small: `small_${id}_${filename}`,
+                    thumbnail: `thumbnail_${id}_${filename}`,
+                }
+            }
+        })
+    }
+
+    async deleteFiles(files) {
+        return files.map(fileName => {
+                blobService.deleteBlobIfExists(containerName, fileName, (err, res) => {
```
Comment (Alan18081) [inline modules/upload/upload.service.js:106]: Please, convert `deleteBlobIfExists` to `Promise` using `promisify` from the default node package `util`

File: modules/upload/upload.service.js [hunk 13]
```diff
@@ -0,0 +1,119 @@
...
+                    medium: `medium_${id}_${filename}`,
+                    small: `small_${id}_${filename}`,
+                    thumbnail: `thumbnail_${id}_${filename}`,
+                }
+            }
+        })
+    }
+
+    async deleteFiles(files) {
+        return files.map(fileName => {
```
Comment (Alan18081) [inline modules/upload/upload.service.js:105]: Please, use here `Promise.all` to return response only when all files are deleted

File: modules/upload/upload.service.js (hunk 2 above)
Comment (Crash1212) [reply modules/upload/upload.service.js:16]: File on Upload type is promise

File: modules/upload/upload.service.js (hunk 3 above)
Comment (Crash1212) [reply modules/upload/upload.service.js]: fixed

File: modules/upload/upload.service.js (hunk 4 above)
Comment (Crash1212) [reply modules/upload/upload.service.js]: fixed

File: modules/upload/upload.service.js (hunk 9 above)
Comment (Crash1212) [reply modules/upload/upload.service.js]: fixed

File: modules/upload/upload.service.js (hunk 6 above)
Comment (Crash1212) [reply modules/upload/upload.service.js]: fixed

File: modules/upload/upload.service.js (hunk 5 above)
Comment (Crash1212) [reply modules/upload/upload.service.js]: fixed

File: modules/upload/upload.service.js (hunk 7 above)
Comment (Crash1212) [reply modules/upload/upload.service.js]: fixed

File: modules/upload/upload.service.js (hunk 8 above)
Comment (Crash1212) [reply modules/upload/upload.service.js]: fixed

File: modules/upload/upload.service.js (hunk 10 above)
Comment (Crash1212) [reply modules/upload/upload.service.js]: fixed

File: modules/upload/upload.service.js (hunk 11 above)
Comment (Crash1212) [reply modules/upload/upload.service.js]: fixed

File: modules/upload/upload.service.js (hunk 12 above)
Comment (Crash1212) [reply modules/upload/upload.service.js:93]: fixed

File: modules/upload/upload.service.js [hunk 14]
```diff
@@ -9,8 +9,36 @@ const containerName = 'images';
...
+        
+        const blobName = imageName;
+        const stream = getStream(buffer);
+        const streamLength = buffer.length;
+
+
+        return await new Promise((resolve, reject) => {
+            blobService.createBlockBlobFromStream(containerName, blobName, stream, streamLength, err => {
+                if (err) {
+                    reject(new Error(err))
```
Comment (Alan18081) [inline modules/upload/upload.service.js:35]: No need to wrap it in `Error`, it is already error instance

File: modules/upload/upload.service.js [hunk 15]
```diff
@@ -9,8 +9,36 @@ const containerName = 'images';
 const getStream = require('into-stream');
 const Jimp = require("jimp");
 const uniqid = require('uniqid');
+const util = require('util');
```
Comment (Alan18081) [inline modules/upload/upload.service.js:12]: Do you use this package for something

File: modules/upload/upload.service.js [hunk 16]
```diff
@@ -9,8 +9,36 @@ const containerName = 'images';
...
 
 class UploadService {
+
+    async uploadResizedImage(size, imageName, image) {
+        const resizedImage = image.resize(size, Jimp.AUTO)
+    
+        const buffer = await new Promise((resolve, reject) => {
+            resizedImage.getBuffer(resizedImage.getMIME(), (err, buffer) => {
+                if(err)
```
Comment (Alan18081) [inline modules/upload/upload.service.js:21]: Please, add curly braces

File: modules/upload/upload.service.js [hunk 17]
```diff
@@ -9,8 +9,36 @@ const containerName = 'images';
...
+                    reject(err)
+                resolve(buffer)
+            })
+        })
+        
+        const blobName = imageName;
```
Comment (Alan18081) [inline modules/upload/upload.service.js:27]: Why do you need this assignment?

File: modules/upload/upload.service.js [hunk 18]
```diff
@@ -35,83 +63,40 @@ class UploadService {
...
             }
         })
     }
 
     async deleteFiles(files) {
-        return files.map(fileName => {
+        return files.map(async fileName => 
+            await new Promise((resolve, reject) =>
                 blobService.deleteBlobIfExists(containerName, fileName, (err, res) => {
                     if(!err){
```
Comment (Alan18081) [inline modules/upload/upload.service.js:94]: If you firstly call `reject` in condition `if (err) {...}` then do it here first too, please

Comment (Crash1212) [reply modules/upload/upload.service.js]: fixed

File: modules/upload/upload.service.js (hunk 17 above)
Comment (Crash1212) [reply modules/upload/upload.service.js]: fixed

File: modules/upload/upload.service.js (hunk 16 above)
Comment (Crash1212) [reply modules/upload/upload.service.js]: fixed

File: modules/upload/upload.service.js (hunk 15 above)
Comment (Crash1212) [reply modules/upload/upload.service.js]: fixed

File: modules/upload/upload.service.js (hunk 14 above)
Comment (Crash1212) [reply modules/upload/upload.service.js]: fixed

**Pull request merged**